src/presenter/trip-event-presenter.js

```diff
@@ -1,67 +1,60 @@
-import { render, replace } from '../framework/render.js';
-import ListEvent from '../view/event-list-view.js';
-import EventsItemView from '../view/events-item-view.js';
-import EventEdit from '../view/event-edit.js';
+import { render } from '../framework/render.js';
 import ListEmpty from '../view/list-empty.js';
-export default class EventPresenter {
+import PointPresenter from './point-presenter.js';
+import EventsListView from '../view/events-list-view.js';
 
-  #eventComponent = new ListEvent();
+export default class EventPresenter {
+  #eventsListView = new EventsListView();
   #points = null;
   #listContainer = null;
   #point = null;
   #offer = null;
   #destination = null;
 
+  #pointPresenter = null;
+
   constructor({listContainer, pointsModel, offersModel, destinationsModel}){
     this.#listContainer = listContainer;
     this.#point = pointsModel;
     this.#offer = offersModel;
     this.#destination = destinationsModel;
-    this.#points = pointsModel.get();
+    this.#points = new Map();
   }
 
-  #renderPoint = (point) => {
-    const pointComponent = new EventsItemView({
-      point,
-      pointDestination : this.#destination.getById(point.destination),
-      pointOffers: this.#offer.getByType(point.type)
-    });
-    // render(pointComponent,this.#eventComponent.element);
 
-    const pointEditComponent = new EventEdit({
-      point,
-      pointDestination : this.#destination.get(),
-      pointOffers: this.#offer.get()
+  #renderPoint(point){
+    this.#pointPresenter = new PointPresenter({
+      eventsListView: this.#eventsListView,
+      point: point,
+      offers: this.#offer,
+      destinations: this.#destination,
     });
+    this.#pointPresenter.init();
 
-    const replacePointToEdit = () => {
-      replace(pointEditComponent, pointComponent);
-    };
-    const replaceEditToPoint = () => {
-      replace(pointComponent, pointEditComponent);
-    };
-
-    pointComponent.setEditHandler(()=>replacePointToEdit());
-    pointEditComponent.setEditHandler(()=>replaceEditToPoint());
-    render(pointComponent,this.#eventComponent.element);
-  };
-
-  #renderEditPoint = (point) => {
-    const pointEditComponent = new EventEdit({
-      point,
-      pointDestination : this.#destination.get(),
-      pointOffers: this.#offer.get()
+    this.#pointPresenter.setOpenEditModeHandler(() => {
+      this.#points.forEach((presentor,pointId)=>{
+        if(point.id !== pointId){
+          presentor.closeEditMode();
+        }
+      });
+    });
+    this.#pointPresenter.setChangeFavoriteHandler(()=>{
+      this.#points.forEach((presentor,pointId) => {
+        if(point.id === pointId){
+          presentor.setValueFavorite();
+        }
+      });
     });
-    render(pointEditComponent,this.#eventComponent.element);
-  };
+    this.#points.set(point.id, this.#pointPresenter);
+  }
 
   init(){
```

Comment:
Лучше достать данные 1 раз и положить их в перемнную, иначе при каждом вызове они, предположим, будут генероиваться снова, хотя тут это не так, но все равно, лучше не надеяться на такое
```suggestion
init() {
    const pointsData = this.#pointsModel.get();
    if (pointsData.length) {
      render(this.#eventsListView, this.#listContainer);

      pointsData.forEach((point) => this.#renderPoint(point));
    } else {
      render(new ListEmpty, this.#listContainer);
    }
  }
```